Sources/Sentry/SentryNSDataTracker.m

```diff
@@ -0,0 +1,186 @@
+{
+    id<SentrySpan> span = [self startTrackingReadingFilePath:path];
+
+    NSData *result = method(path);
+
+    if (span != nil) {
+        [self finishTrackingNSData:result span:span];
+    }
+
+    return result;
+}
+
+- (NSData *)measureNSDataFromFile:(NSString *)path
+                          options:(NSDataReadingOptions)readOptionsMask
+                            error:(NSError **)error
+                           method:(NSData * (^)(NSString *, NSDataReadingOptions, NSError **))method
+{
+    id<SentrySpan> span = [self startTrackingReadingFilePath:path];
+
+    NSData *result = method(path, readOptionsMask, error);
+
+    if (span != nil) {
+        [self finishTrackingNSData:result span:span];
+    }
+
+    return result;
+}
+
+- (NSData *)measureNSDataFromURL:(NSURL *)url
+                         options:(NSDataReadingOptions)readOptionsMask
+                           error:(NSError **)error
+                          method:(NSData * (^)(NSURL *, NSDataReadingOptions, NSError **))method
+{
+
+    // We dont track reads from a url that is not a file url
+    // because these reads are handled by NSURLSession and
+    // SentryNetworkTracker will create spans in these cases.
+    if (![url.scheme isEqualToString:NSURLFileScheme])
+        return method(url, readOptionsMask, error);
+
+    id<SentrySpan> span = [self startTrackingReadingFilePath:url.path];
+
+    NSData *result = method(url, readOptionsMask, error);
+
+    if (span != nil) {
+        [self finishTrackingNSData:result span:span];
+    }
+
+    return result;
+}
+
+- (nullable id<SentrySpan>)spanForPath:(NSString *)path
+                             operation:(NSString *)operation
+                                  size:(NSUInteger)size
+{
+    @synchronized(self) {
+        if (!self.isEnabled || ![self shouldTrackPath:path])
+            return nil;
+    }
+
+    __block id<SentrySpan> ioSpan;
+    [SentrySDK.currentHub.scope useSpan:^(id<SentrySpan> _Nullable span) {
+        ioSpan = [span startChildWithOperation:operation
+                                   description:[self transactionDescriptionForFile:path
+                                                                          fileSize:size]];
+    }];
+
+    [ioSpan setDataValue:path forKey:@"file.path"];
+    return ioSpan;
+}
+
+- (nullable id<SentrySpan>)startTrackingWritingNSData:(NSData *)data filePath:(NSString *)path
+{
+    return [self spanForPath:path operation:SENTRY_FILE_WRITE_OPERATION size:data.length];
+}
+
+- (nullable id<SentrySpan>)startTrackingReadingFilePath:(NSString *)path
+{
+    return [self spanForPath:path operation:SENTRY_FILE_READ_OPERATION size:0];
+}
+
+- (void)finishTrackingNSData:(NSData *)data span:(id<SentrySpan>)span
+{
+    [span setDataValue:[NSNumber numberWithUnsignedInteger:data.length] forKey:@"file.size"];
+    [span finish];
+}
+
+- (BOOL)shouldTrackPath:(NSString *)path
```

Comment:
`l`: The method name `shouldTrackPath` is a bit confusing to me. We only check if the path is from the SentrySDK. We could rename it to something like: `isSentryPath` and remove the `!` on the return line.

Reply:
For now, is the only rule I can think of. But maybe Im missing something, or new rules regarding the file path will appear. Maybe a user list in SentryOptions of paths or files that should not be tracked.

Reply:
If new rules appear, we can always rename the method at it's not public API.